src/main/kotlin/gropius/model/template/BaseTemplate.kt

```diff
@@ -0,0 +1,73 @@
+package gropius.model.template
+
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+import com.expediagroup.graphql.generator.annotations.GraphQLIgnore
+import com.expediagroup.graphql.generator.annotations.GraphQLType
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.JsonNodeFactory
+import gropius.model.common.NamedNode
+import io.github.graphglue.model.Direction
+import io.github.graphglue.model.DomainNode
+import io.github.graphglue.model.Node
+import io.github.graphglue.model.NodeRelationship
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.data.annotation.Transient
+import org.springframework.data.neo4j.core.schema.CompositeProperty
+
+/**
+ * @param templateFieldSpecifications the schema for all templated fields defined by this template
+ */
+@DomainNode
+@GraphQLDescription(
+    """Base type for both Template and SubTemplate.
+    Defines templated fields with specific types (defined using JSON schema).
+    """
+)
+abstract class BaseTemplate<T : Node, S : BaseTemplate<T, S>>(
```

Comment:
Is it possible in kotlin to enforce, that T ALSO needs to be of type TemplatedNode in addition to Node?

Reply:
possible? yes 
does it provide any benefits in our case? I'm note sure 
does it cause any harm? no 
therefore added it with 24dc068aa37bdd6913a43d6e2c26e558ba8265e5

Reply:
Nice